src/Uno.UI.Runtime.Skia.Gtk/GtkCoreWindowExtension.cs

```diff
@@ -436,48 +462,88 @@ private static PointerDevice ToPointerDevice(Gdk.Device sourceDevice)
 				case InputSource.Eraser:
 				case InputSource.TabletPad: // the device is a "pad", a collection of buttons, rings and strips found in drawing tablets.
 				case InputSource.Cursor: // the device is a graphics tablet “puck” or similar device.
-					return PointerDevice.For(PointerDeviceType.Pen);
+					return PointerDeviceType.Pen;
 
 				case InputSource.Touchscreen:
-					return PointerDevice.For(PointerDeviceType.Touch);
+					return PointerDeviceType.Touch;
 
 				case InputSource.Mouse:
 				default:
-					return PointerDevice.For(PointerDeviceType.Mouse);
+					return PointerDeviceType.Mouse;
 			}
 		}
 
 		private static bool IsPressed(ModifierType state, ModifierType mask, PointerUpdateKind update, PointerUpdateKind pressed, PointerUpdateKind released)
 			=> update == pressed || (state.HasFlag(mask) && update != released);
 
-		private void RaisePointerExited(PointerEventArgs ptArgs)
+
+#if TRACE_NATIVE_POINTER_EVENTS
+		private void RaisePointerEntered(PointerEventArgs ptArgs, [CallerMemberName] string caller = null)
 		{
+			Trace($"[Entered] ({caller}) {ptArgs}");
```

Comment:
`[Conditional("TRACE_NATIVE_POINTER_EVENTS")]` ?

Reply:
It was my first implementation, but as I wanted to include the `[CallerMemberName]` in the log, I preferred to just switch the whole `RaiseXXX` methods to make sure that don't have any use less overhead in those methods that can be on a "hot path".